File: yunva17/프로그래머스/Lv2/타겟_넘버.js
const solution = (numbers, target) => {
  let answer = 0;

  const dfs = (index, sum) => {
    if (index === numbers.length) {
      if (sum === target) {
        answer++;
      }
      return;
    }

    dfs(index + 1, sum + numbers[index]);
    dfs(index + 1, sum - numbers[index]);
  };

  dfs(0, 0);

  return answer;
};

File: pbkdpwls/DFS,BFS/타겟넘버.java
class Solution {
    int answer = 0;
    public int solution(int[] numbers, int target) {

        dfs(0, 0, target, numbers);

        return answer;
    }
    public void dfs(int depth, int temp, int target, int[] numbers) {
        if (temp==target && depth==numbers.length) {
            answer++;
        }
        if(depth<numbers.length) {
            dfs(depth+1,temp-numbers[depth], target, numbers);
            dfs(depth+1,temp+numbers[depth], target, numbers);
        }
    }
}